File: projects/m1/036-gregorian-date-to-ordinal-date/solutions/M.GIOVARRUSCIO/GREGORIAN-DATE-TO-ORDINAL-DATE.py
"""
An ordinal date consists of a year and a day within it, both of which are integers.
The year can be any year in the Gregorian calendar
while the day within the year ranges from one, which represents January 1,
through to 365 (or 366 if the year is a leap year) which represents December 31.
Ordinal dates are convenient when computing differences between dates
that are a specific number of days (rather than months.
For example, ordinal dates can be used to easily determine
whether a customer is within a 90 day return period,
the sell-by date for a food-product based on its production date, and the due date for a baby.

Write a function named ordinalDate that takes three integers as parameters.

These parameters will be a day, month and year respectively.
The function should return the day within
the year for that date as its only result.
Create a main program that reads a day, month and year from the user
and displays the day within the year for that date.
Your main program should only run when your file has not been imported into another program.

"""
def leapyear(anno):
    if (anno % 400 == 0) or ((anno % 4 == 0) and (anno % 100 != 0)):
        return True
    else:
        return False
def OrdinaData(g, m, a):
    day = g
    for i in range(1,m):
        if i in [1, 3, 5, 6, 7, 10, 12]:
            day += 31
        elif i in [11, 4, 6, 9]:
            day += 30
        elif i == 2 and leapyear(a):
            day += 29
        elif i == 2 and not leapyear(a):
            day += 28
    print(f"Date ordinal is {day} for date {g}-{m}-{a}")


if __name__ == '__main__':
    gg = int(input("day:"))
    mm = int(input("mounth:"))
    aa = int(input("year:"))
    OrdinaData(gg, mm, aa)

File: projects/m1/036-gregorian-date-to-ordinal-date/solutions/alessiacaporaso/gregorian-date-to-ordinal-date.js
/*
Write a function named ordinalDate that takes three integers as parameters.
These parameters will be a day, month and year respectively. 
The function should return the day within the year for that date as its only result.
Create a main program that reads a day, month and year from the user and displays the day within the year for that date. 
Your main program should only run when your file has not been imported into another program.
*/

function isLeap(year) {
  if (year % 400 === 0 || (year % 4 === 0 && year % 100 !== 0)) {
    return true;
  } else {
    return false;
  }
}
function ordinalDate(day, month, isLeap) {
  let numberDay = [];
  // create a variable with the number od day in a month    
  for (let i = 1; i < month; i++) {
    switch (i) {
      case 4:
      case 6:
      case 9:
      case 11:
        numberDay.push(30);
        break;
      case 2:
        if (isLeap === true) {
          numberDay.push(29);
        } else {
          numberDay.push(28);
        }
        break;
      default:
        numberDay.push(31);
        break;
    }
  }
  // sum all day in a months
  let sumDayInMonth = 0;
  for (let item of numberDay) {
    sumDayInMonth += item;
  }

  const calcordinalDate = sumDayInMonth + day;
  alert("The ordinal date is: " + calcordinalDate);
}

const date = prompt("Please, enter a date", "10/12/2022");
const splitDate = date.split("/");
const userDay = parseInt(splitDate[0]);
const userMonth = parseInt(splitDate[1]);
const userYear = parseInt(splitDate[2]);

ordinalDate(userDay, userMonth);
isLeap(userYear);
